tiledb/api/c_api/query_aggregate/test/unit_capi_query_aggregate.cc

```diff
@@ -0,0 +1,361 @@
+  CHECK(tiledb_channel_create_operator_sum(ctx, &sum_operator) == TILEDB_OK);
+
+  // nullptr context
+  tiledb_query_channel_t* default_channel;
+  tiledb_channel_operation_t* count_op;
+  tiledb_channel_operator_t* o;
+  CHECK(
+      tiledb_channel_create_operator_count(nullptr, &o) ==
+      TILEDB_INVALID_CONTEXT);
+  CHECK(
+      tiledb_channel_create_operator_sum(nullptr, &o) ==
+      TILEDB_INVALID_CONTEXT);
+  CHECK(
+      tiledb_query_get_default_channel(nullptr, query, &default_channel) ==
+      TILEDB_INVALID_CONTEXT);
+  CHECK(
+      tiledb_channel_create_operation_field(
+          nullptr, query, count_operator, "a", &count_op) ==
+      TILEDB_INVALID_CONTEXT);
+  CHECK(
+      tiledb_channel_add_aggregate(
+          nullptr, default_channel, "Count", count_op) ==
+      TILEDB_INVALID_CONTEXT);
+
+  // nullptr query
+  CHECK(
+      tiledb_query_get_default_channel(ctx, nullptr, &default_channel) ==
+      TILEDB_ERR);
+  CHECK(
+      tiledb_channel_create_operation_field(
+          ctx, nullptr, count_operator, "a", &count_op) == TILEDB_ERR);
+
+  // nullptr operator
+  CHECK(
+      tiledb_channel_create_operation_field(
+          ctx, query, nullptr, "a", &count_op) == TILEDB_ERR);
+  CHECK(tiledb_channel_operator_free(nullptr) == TILEDB_ERR);
+  tiledb_channel_operator_t* nulloperator = NULL;
+  CHECK(tiledb_channel_operator_free(&nulloperator) == TILEDB_ERR);
+
+  // nullptr input field
+  CHECK(
+      tiledb_channel_create_operation_field(
+          ctx, query, sum_operator, nullptr, &count_op) == TILEDB_ERR);
+
+  // nullptr input field allowed for count
+  CHECK(
+      tiledb_channel_create_operation_field(
+          ctx, query, count_operator, nullptr, &count_op) == TILEDB_OK);
+
+  // nullptr channel
+  CHECK(
+      tiledb_channel_add_aggregate(ctx, nullptr, "Count", count_op) ==
+      TILEDB_ERR);
+  CHECK(tiledb_query_channel_free(nullptr) == TILEDB_ERR);
+  tiledb_query_channel_t* nullchannel = NULL;
+  CHECK(tiledb_query_channel_free(&nullchannel) == TILEDB_ERR);
+
+  // nullptr output field
+  CHECK(
+      tiledb_channel_add_aggregate(ctx, default_channel, nullptr, count_op) ==
+      TILEDB_ERR);
+
+  // nullptr operation
+  CHECK(
+      tiledb_channel_add_aggregate(ctx, default_channel, "Count", nullptr) ==
+      TILEDB_ERR);
+  CHECK(tiledb_channel_operation_free(nullptr) == TILEDB_ERR);
+  tiledb_channel_operation_t* nullop = NULL;
+  CHECK(tiledb_channel_operation_free(&nullop) == TILEDB_ERR);
+
+  // Clean up
+  CHECK(tiledb_channel_operator_free(&sum_operator) == TILEDB_OK);
+  CHECK(tiledb_channel_operator_free(&count_operator) == TILEDB_OK);
+  CHECK(tiledb_array_close(ctx, array) == TILEDB_OK);
+  tiledb_array_free(&array);
+  tiledb_query_free(&query);
+}
+
+TEST_CASE_METHOD(
+    QueryAggregateFx,
+    "C API: Query aggregates COUNT test",
+    "[capi][query_aggregate][count]") {
+  std::string array_name = temp_dir_ + "queryaggregate_array";
```

Comment:
I haven't looked too much at other tests in the c_api folder... What is normally the goal of these tests? Do they try to test everything extensively of just test the API for invalid parameters fully then have very basic success cases?